File: z-clients/javascript/src/models/rateLimitCreateNamespaceOut.ts
// this file is @generated

export interface RateLimitCreateNamespaceOut {
    name: string;
    created: Date;
    updated: Date;
}

export const RateLimitCreateNamespaceOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): RateLimitCreateNamespaceOut {
        return {
            name: object['name'],
            created: new Date(object['created']),
            updated: new Date(object['updated']),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: RateLimitCreateNamespaceOut): any {
        return {
            'name': self.name,
            'created': self.created.getTime(),
            'updated': self.updated.getTime(),
        };
    }
}

File: z-clients/rust/src/models/rate_limit_create_namespace_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct RateLimitCreateNamespaceOut {
    pub name: String,

    pub created: u64,

    pub updated: u64,
}

impl RateLimitCreateNamespaceOut {
    pub fn new(name: String, created: u64, updated: u64) -> Self {
        Self {
            name,
            created,
            updated,
        }
    }
}
